qemu/tests/qemu_guest_agent.py

```diff
@@ -256,7 +264,8 @@ def run_once(self, test, params, env):
             vm = self.env.get_vm(params["main_vm"])
             vm.verify_alive()
             self.vm = vm
-            self.gagent_verify(self.params, self.vm)
+            if params.get("gagent_serial_type"):
+                self.gagent_verify(self.params, self.vm)
```

Comment:
If you want to distinguish the `hotplug` one and the `non-hotplug` one, just say:
```python
if params.get("hotplug_gagent", "no") == "no":
    self.gagent_verify(self.params, self.vm)
```